lucene/test-framework/src/java/org/apache/lucene/util/fst/FSTTester.java

```diff
@@ -95,7 +95,7 @@ private static BytesRef toBytesRef(IntsRef ir) {
     return br;
   }
 
-  static String getRandomString(Random random) {
+  public static String getRandomString(Random random) {
```

Comment:
Looks like this should be publicly accessible for tests in any modules?

Reply:
Don't know what it actually does but randomizedtesting has a number of utility methods for random strings:

http://labs.carrotsearch.com/download/randomizedtesting/2.0.0/api/randomizedtesting-runner/com/carrotsearch/randomizedtesting/generators/RandomStrings.html

Reply:
Yes, they look similar except for that Lucene's TestUtil selects different codepoint range (`blockStarts` and `blockEnds`) to generate random String instance  each time. I think both one is okay for the test, what do you think?
 
o.a.l.util.TestUtil.randomRealisticUnicodeString()
```
  /** Returns random string of length between min and max codepoints, all codepoints within the same unicode block. */
  public static String randomRealisticUnicodeString(Random r, int minLength, int maxLength) {
    final int end = nextInt(r, minLength, maxLength);
    final int block = r.nextInt(blockStarts.length);
    StringBuilder sb = new StringBuilder();
    for (int i = 0; i < end; i++)
      sb.appendCodePoint(nextInt(r, blockStarts[block], blockEnds[block]));
    return sb.toString();
  }
```

com.carrotsearch.randomizedtesting.generators.UnicodeGenerator.ofCodePointsLength()
```
  public String ofCodePointsLength(Random r, int minCodePoints, int maxCodePoints) {
    int length = RandomNumbers.randomIntBetween(r, minCodePoints, maxCodePoints);
    int [] chars = new int [length];
    for (int i = 0; i < chars.length; i++) {
      int v = RandomNumbers.randomIntBetween(r, 0, CODEPOINT_RANGE);
      if (v >= Character.MIN_SURROGATE)
        v += SURROGATE_RANGE;
      chars[i] = v;
    }
    return new String(chars, 0, chars.length);
  }
```

Reply:
Not sure whether the difference matters. It shouldn't at first glance.

Reply:
If there's no good reason to change, I'd like to keep the original test code here and make the test utility method publicly accessible (the change shouldn't hurt anything, to me) :)

Reply:
If you're going to make this method public... I think even a tiny bit of documentation would be helpful.  Like maybe recommend callers consider the methods Dawid mentioned in RandomizedTesting _instead_.  After all, we're only making this public for some test in misc to call it.  We aren't signaling to all of Lucene, Solr, and any of the plugin writers, that we've got some awesome random string generating that everyone should use :-)

FWIW I prefer that we maintain less code and use whatever's in RandomizedTesting when possible, but I won't stand in the way of your judgement here.

Reply:
I'd use RandomStrings instead of making the FSTTester method public and adding extra documentation for it (I prefer only one recommended way to implement one thing), if we can safely switch them. 

I checked the `FSTTester.getRandomString()` method if we can safely replace it with `RandomStrings.randomUnicodeOfCodepointLength()`. It seems there is a small, but intentional difference (it generates strings that consists of characters within the range of codepoint 97('a') to 102('f'), with a 50% possibility)... not fully sure the details, but it somewhat serves for testing FST?

https://github.com/apache/lucene-solr/blob/master/lucene/test-framework/src/java/org/apache/lucene/util/fst/FSTTester.java#L98-L122

Reply:
RandomStrings might be okay here but I'm not so confident about switching to it because of the difference... i aded a note to the method's javadoc. https://github.com/apache/lucene-solr/pull/2064/commits/a0d97b595f8d01f81db6d1f068aa8ae11d47887e